### PR TITLE
fix jwt-cpp on_test

### PR DESCRIPTION
**Pull request by TTopoo**
check package:config("picojson") before on_test

### CODE REVIEW

File: packages/j/jwt-cpp/xmake.lua [hunk 1]
```diff
@@ -31,11 +31,13 @@ package("jwt-cpp")
...
-        assert(package:check_cxxsnippets({test = [[
-            #include <jwt-cpp/jwt.h>
-            void test() {
-                std::string token;
-                auto decoded = jwt::decode(token);
-            }
-        ]]}, {configs = {languages = "c++11"}}))
+        if package:config("picojson") then
+            assert(package:check_cxxsnippets({test = [[
+                #include <jwt-cpp/jwt.h>
```
[the remark this answers is not in the document]
Comment (TTopoo) [reply packages/j/jwt-cpp/xmake.lua:36]: image
If I use add_requires("jwt-cpp", {configs = {picojson = false}}), the situation shown in the figure will appear when installing the package

[the remark this answers is not in the document]
Comment (TTopoo) [reply packages/j/jwt-cpp/xmake.lua:36]: Yes, I use nlohmann-json by add_requires in my project code, I just want to be able to install jwt after disabling picojson.
If the programmer actively turns off this picojson config, it means that he is consciously using other json libraries.

[the remark this answers is not in the document]
Comment (star-hengxing) [reply packages/j/jwt-cpp/xmake.lua:36]: Need refactor the package

**Comment by star-hengxing** [comment]
No issue? I merged

**Pull request merged**